pkg/kubelet/certificate/transport.go

```diff
@@ -77,16 +83,35 @@ func updateTransport(stopCh <-chan struct{}, period time.Duration, clientConfig
 		conns:  make(map[*closableConn]struct{}),
 	}
 
+	lastCertAvailable := time.Now()
 	lastCert := clientCertificateManager.Current()
 	go wait.Until(func() {
 		curr := clientCertificateManager.Current()
-		if exitIfExpired && curr != nil && time.Now().After(curr.Leaf.NotAfter) {
-			if clientCertificateManager.ServerHealthy() {
-				glog.Fatalf("The currently active client certificate has expired and the server is responsive, exiting.")
+
+		if exitAfter > 0 {
+			now := time.Now()
+			if curr == nil {
```

Comment:
Is this the same without the if nesting? 
```golang
if !clientCertificateManager.ServerHealthy() {
  glog.Errorf("server down, trying again")
  return
}
if curr != nil {
  lastCertAvailable = now
}
if now.After(lastCertAvailable.Add(exitAfter)) {
  glog.Fatalf("It has been %s since a valid client cert was found and the server is responsive, exiting.", exitAfter)
}
if now.After(curr.Leaf.NotAfter) {
  glog.Fatalf("The currently active client certificate has expired and the server is responsive, exiting.")
}
```

Reply:
Not 100%, although it may just be better if this is easier to reason about than if the exact message is precise.

Reply:
I'm somewhat hesitant to change the logic to look simpler but give worse messages.  I agree this particular loop is ugly.